cli/fmt.rs

```diff
@@ -47,62 +47,42 @@ pub async fn format(
   }
   let config = get_config();
   if check {
-    check_source_files(config, target_files).await
+    check_source_files(config, &target_files)
   } else {
-    format_source_files(config, target_files).await
+    format_source_files(config, &target_files)
   }
 }
 
-async fn check_source_files(
+fn check_source_files(
   config: dprint::configuration::Configuration,
-  paths: Vec<PathBuf>,
+  paths: &[PathBuf],
 ) -> Result<(), ErrBox> {
   let not_formatted_files_count = Arc::new(AtomicUsize::new(0));
-  let formatter = Arc::new(dprint::Formatter::new(config));
+  let has_error = Arc::new(AtomicBool::new(false));
+  let output_lock = Arc::new(Mutex::new(())); // prevent threads outputting at the same time
+  let formatter = dprint::Formatter::new(config);
 
-  // prevent threads outputting at the same time
-  let output_lock = Arc::new(Mutex::new(0));
-
-  run_parallelized(paths, {
-    let not_formatted_files_count = not_formatted_files_count.clone();
-    move |file_path| {
-      let file_text = read_file_contents(&file_path)?.text;
-      let r = formatter.format_text(&file_path, &file_text);
-      match r {
-        Ok(formatted_text) => {
-          if formatted_text != file_text {
-            not_formatted_files_count.fetch_add(1, Ordering::SeqCst);
-            let _g = output_lock.lock().unwrap();
-            match diff(&file_text, &formatted_text) {
-              Ok(diff) => {
-                println!();
-                println!(
-                  "{} {}:",
-                  colors::bold("from"),
-                  file_path.display().to_string()
-                );
-                println!("{}", diff);
-              }
-              Err(e) => {
-                eprintln!(
-                  "Error generating diff: {}",
-                  file_path.to_string_lossy()
-                );
-                eprintln!("   {}", e);
-              }
-            }
-          }
-        }
-        Err(e) => {
-          let _g = output_lock.lock().unwrap();
-          eprintln!("Error checking: {}", file_path.to_string_lossy());
-          eprintln!("   {}", e);
+  run_parallelized(paths, |file_path| {
+    let r = check_file(&formatter, &file_path, output_lock.clone());
+    match r {
+      Ok(changed) => {
+        if changed {
+          not_formatted_files_count.fetch_add(1, Ordering::SeqCst);
         }
       }
-      Ok(())
+      Err(e) => {
+        has_error.store(true, Ordering::Relaxed);
+        let _g = output_lock.lock().unwrap();
+        eprintln!("Error checking: {}", file_path.to_string_lossy());
+        eprintln!("   {}", e);
+      }
     }
-  })
-  .await?;
+  });
+
+  let has_error = has_error.load(Ordering::Relaxed);
+  if has_error {
+    std::process::exit(1);
+  }
```

Comment:
I'm just realizing that some of the code in here and in `format_source_files` could be extracted out and pushed down to `run_parallelized` (all the error handling inside the delegate too). Let me know if I should do that.

Reply:
@dsherret I think it's fine for now